code/modules/hydroponics/biogenerator.dm

```diff
@@ -19,6 +19,42 @@
 /obj/machinery/biogenerator/Initialize(mapload)
 	. = ..()
 	stored_research = new /datum/techweb/specialized/autounlocking/biogenerator
+	register_context()
+
+/obj/machinery/biogenerator/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	if(!isliving(user))
+		return .
+
+	if(!Adjacent(user))
+		return .
+
+	switch (held_item?.tool_behaviour)
+		if (TOOL_SCREWDRIVER)
+			context[SCREENTIP_CONTEXT_LMB] = panel_open ? "Close panel" : "Open panel"
+			return CONTEXTUAL_SCREENTIP_SET
+		if (TOOL_CROWBAR)
+			if (panel_open)
+				context[SCREENTIP_CONTEXT_LMB] = "Remove electronics"
+				return CONTEXTUAL_SCREENTIP_SET
+
+	if (istype(held_item, /obj/item/food/grown/ ))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert product"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/storage/bag/plants))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert products"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/reagent_containers/glass ))
+		context[SCREENTIP_CONTEXT_LMB] =  beaker ? "Swap container" : "Insert container"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/ ))
+		return .
+		
+	context[SCREENTIP_CONTEXT_LMB] = "Open UI"
```

Comment:
```suggestion
	context[SCREENTIP_CONTEXT_LMB] = "Use"
```
I don't think 'Open UI' fits thematically, for non-silicons at least.

Reply:
Considering the Screentip is an OOC information for the player, when I mean to "Open the interface", it's about the biogenerator's window with the controls.
Also IC I suppose the biogenerator likely has a terminal screen or similar, with a "Graphical User Interface" programmed to handle the controls. 
In this last situation you could then "Turn on the interface" or whatever approach. But if screentips are more indicators for the gameplay controls and must ideally stay short, I sticked to "Open UI" because it's what it does. You open the user interface, and you close when you are away/not using it anymore.

The term "Use" is vaguely generic for a left click : you can "use" an airlock, you can "use" a deskbell, you can "use" a screwdriver on an airlock. In this context, all atoms would have a "LMB : Use" then.